.editorconfig

```diff
@@ -20,24 +17,13 @@ csharp_style_expression_bodied_constructors = false:silent
 csharp_style_expression_bodied_operators = false:silent
 csharp_indent_labels = one_less_than_current
 
-[*.sln]
-indent_style = tab
-
 # Xml project files
 [*.{config,csproj,njsproj,targets,vcxitems,vcxproj,vcxproj.filters}]
 indent_size = 2
-end_of_line = crlf
-insert_final_newline = false
```

Comment:
I'd say we should leave insert_final_newline = false in. These are specific VS formats. VS never adds final new lines and removes them when found. It is safe and reasonable to leave this setting in the editorconfig, so other code editors won't accidentially insert final new lines when it is their default.

Reply:
> VS never adds final new lines and removes them when found

I don't see VS removing final new lines.

> It is safe and reasonable to leave this setting in the editorconfig, so other code editors won't accidentially insert final new lines when it is their default

The right way to achieve this would be 

`insert_final_newline = unset
`

The problem with insert_final_newline is that some editors interpret it as instruction to remove a final newline. Setting it to false sounds like telling the editor to do nothing, but that's not always the result you get. The right way to have an editor do nothing wrt some rule is its absense. This is what unset achieves.

Reply:
> > VS never adds final new lines and removes them when found
> 
> I don't see VS removing final new lines.

I guess what you mean is when making changes in the project designer in old-style (non-SDK) project, where the project desginer writes it out like this.
When you edit a csproj file for text editing (which is the same as what people with other editors would be doing), VS doesn't remove final line breaks.
For SDK style projects, it doesn't even do this when making changes with the project designer.